doc/radosgw/zone-features.rst

```diff
@@ -65,6 +67,19 @@ Once this feature is enabled on all zonegroups in the realm, a background proces
 will convert existing v1 topics and bucket notifications into their v2 format.
 
 
+.. _feature_cross_zonegroup_sync_inc:
+
+cross-zonegroup-sync-inc
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This feature enables bucket replication between zonegroups to start incrementally,
+rather than performing a full sync of existing objects.
+When enabled, replication will only apply to changes and new objects created after
+replication is enabled. This is especially useful for buckets with a large number
+of existing objects, where you may not want to wait for a full sync to complete.
+It reduces the initial load and accelerates the start of the replication process.
```

Comment:
this behavior should apply to bucket replication policy in general, whether it's same- or cross-zonegroup

worth mentioning that this is the aws default for bucket replication policy after their deprecation of  `ExistingObjectReplication`. that's why it's enabled by default on new deployments, but requires existing deployments to opt in

because aws uses the term `ExistingObjectReplication`, maybe we should call this feature something like `skip-existing-object-replication-policy`?

Reply:
>> i'm referring to 'normal bucket sync' where the destination bucket is the same as the source bucket, just on another zone in the zonegroup. the kind you get by default in multisite without any bucket replication policy
>> i've been arguing that this replication should always be symmetric within the zonegroup, and for deprecation of features like sync policy and 'bucket sync disable' that break this symmetry. 
>
> this behavior should apply to bucket replication policy in general, whether it's same- or cross-zonegroup

ok so I guess that was my misunderstanding from the latter quote, you meant we should deprecate/block [PutBucketReplication API](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketReplication.html) calls or `radosgw-admin bucket sync group` commands followed by `--bucket` when it's about the same zone (whether the bucket is the same or the zone) so we can always guarantee the symmetry between zones in a zonegroup.

Reply:
> can we find a way to make this specific to bucket replication policy instead of cross-zonegroup?

In that case, how are we going to tackle the existing policies pointing to the same zonegroup but with different zones or filters? are we going to release a breaking change in favor of the new cross-zonegroup feature or am I confused?

Reply:
when i talk about symmetry between zones, i'm referring to the contents of a specific bucket. in the absense of sync policy or bucket replication policy, a given bucket should (eventually) have the same contents on each zone

bucket replication policy doesn't break this symmetry unless the source and destination bucket are the same. so i support your commit https://github.com/ceph/ceph/commit/888ddcbde7717ead3322077f18776dafcfed98b3 to reject such policies 👍

Reply:
i've been slow to draft a design, but my general idea for cross-zonegroup replication is to add new primitives to enable/allow/forbid the sync relationships between zonegroups, similar to how existing sync policy treats zones

then we could deprecate the existing zone-level sync policy and other features that break per-bucket symmetry (like 'bucket sync disable' and our Zone extension to PutBucketReplication)